lib/src/serializers/json_api.dart

```diff
@@ -43,6 +42,9 @@ class JsonApiSerializer implements Serializer {
       if (withIncluded) {
         jsonMap['included'] = jsonApiDoc.included;
       }
+      if (jsonApiDoc.id != null) {
```

Comment:
@algodave looks like a valid fix, what do you think?

Reply:
@Bringoff Totally agree, thanks @nicobritos 👍  🙂